### ISSUE TITLE
Focus structure modification

### ISSUE THREAD

**Issue by blueskywwc** [Nebula4869/YOLOv5-ncnn-arm#1]
Hello, why should the Focus structure be modified to an interpolation method: torch.nn.functional.interpolate(x, scale_factor=0.5), I saw some documents suggest to modify it to: return self.conv(torch.cat([x,x ,x,x], 1)), is there any difference or advantage between the two, thank you!

**Comment by Nebula4869** [comment]
I think the two implementations are identical except for the different interfaces that are called




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "Nebula4869/YOLOv5-ncnn-ARM"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年12月2日(星期三) 下午2:46
收件人:&nbsp;"Nebula4869/YOLOv5-ncnn-ARM"<YOLOv5-ncnn-ARM@noreply.github.com&gt;;
抄送:&nbsp;"Subscribed"<subscribed@noreply.github.com&gt;;
主题:&nbsp;[Nebula4869/YOLOv5-ncnn-ARM] Foucs structure modification (#1)





 
Hello, why should the Foucs structure be modified to an interpolation method: torch.nn.functional.interpolate(x, scale_factor=0.5), I saw some documents suggest to modify it to: return self.conv(torch.cat([x,x ,x,x], 1)), is there any difference or advantage between the two, thank you!

**Comment by blueskywwc** [comment]
Thank you for your reply. I just tested two methods. The model size and inference time of the interpolation method are consistent with yolov5s.pt. The model of the other method increases, and the inference time doubles (the actual reason is that the size of the feature map is not reduced. , And slice reduces the feature map by half), so I think the interpolation method is more reasonable.
